app/app.go

```diff
@@ -58,6 +60,7 @@ func NewApp(name string, cfg *config.Config, handler http.Handler, debugMode boo
 	a.Handler = handler
 	a.debugMode = debugMode
 	a.signals = make(chan os.Signal, 1)
+	a.Throttler = middleware.NewThrottler(a.Config.API.MaxParallelReq)
```

Comment:
Before this call is made we should ensure that we are happy that the specified value is sane... Whether in the mainline code path here, or as part of the config parsing, or somewhere in between ;-)

Given that the default value for an integer field is going to be 0, if the config value is not specified this would result in setting up throttling with 0 simultaneous connections allowed... Not something we want to do, and may lead to a runtime failure...

Should also be a scenario we test for...